modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -985,6 +989,76 @@ public void testAddObjectAction() throws Exception {
 		_objectActionLocalService.deleteObjectAction(systemObjectAction);
 	}
 
+	@Test
+	public void testAddObjectActionInSystemObjectDefinitionWithConditionExpression()
+		throws Exception {
+
+		SystemObjectDefinitionManager userSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager("User");
+
+		ObjectDefinition userObjectDefinition =
+			_objectDefinitionLocalService.fetchSystemObjectDefinition(
+				TestPropsValues.getCompanyId(), "User");
+
+		ObjectFieldTestUtil.addCustomObjectField(
```

Comment:
@JhossephG make sure to delete the custom object field at the end of this test, otherwise it will affect other integration tests that are using the User system object definition.